src/config.cpp

```diff
@@ -659,14 +660,35 @@ void ParseConfig(const fs::path& filename, Config& config) {
   }
 }
 
-Config::Config(const fs::path& path) : config_path{path} {
+Config::Config(const fs::path& path, const RuntimeSettings* settings) : config_path{path} {
   ParseConfig(path / "genai_config.json", *this);
 
   if (model.context_length == 0)
     throw std::runtime_error("model context_length is 0 or was not set. It must be greater than 0");
 
   if (search.max_length == 0)
     search.max_length = model.context_length;
+
```

Comment:
From how it looks, RuntimeSettings is really just extra settings that can overwrite the genai_config.json settings?
Why can't this be passed as a json string that gets parsed after the original file does? Then it should "just work" and any new options would be available in both automatically.

Reply:
The data "dawn proc table" that we want to use is a runtime data. It is a pointer to the Dawn's API table, which is quite similar to what `OrtApiBase::GetApi()` returns.

The actual use case is, user's application already loads the dawn DLL and can get the proc table. This memory address need to be passed to ORT so that ORT WebGPU EP can use the loaded instance of dawn.

Reply:
Right, I was meaning stuff that was passed at runtime, not written to the config. It looks like the table is just some strings (or am I misunderstanding the to_string part of the parsing?)

Can you show an example of what the data that is being passed in?

Reply:
The actual data "dawn proc table" is an address to a set of API function pointers, pretty much similar to what `OrtApiBase::GetApi()` does. Any other Dawn API requires this to work.

Specifically, User's application need to call [`dawn::native::GetProcs()`](https://github.com/google/dawn/blob/main/src/dawn/native/DawnNative.cpp#L52-L54) to get the proc table, pass the pointer to onnxruntime which uses [`dawnProcSetProcs()`](https://github.com/google/dawn/blob/main/include/dawn/dawn_proc.h#L44) to set the value.

Reply:
The `std::to_string` part is because there is no way in to pass EP specific non-string data to ORT using existing ORT APIs. `OrtAppendExecutionProvider()` supports only a string-to-string kvp as EP specific data.

Reply:
I see, it's just that the config structure should directly map to the json, and this breaks that. So trying to figure out if there's a way to work within this design vs adding exceptions.